tst/EnergyPlus/unit/SolarShading.unit.cc

```diff
@@ -3123,3 +3123,274 @@ TEST_F(EnergyPlusFixture, SolarShading_TestSurfsPropertyViewFactor)
     EXPECT_DOUBLE_EQ(0.40, win_Surface.ViewFactorSkyIR);
     EXPECT_DOUBLE_EQ(0.40, win_Surface.ViewFactorGroundIR);
 }
+
+TEST_F(EnergyPlusFixture, SolarShadingTest_CTRANS)
```

Comment:
At one point I almost went with a solution that involved transforming the vertices coordinates to a new system that was aligned on the surface's plane and going through some points. I noticed the `lcsx` and co vectors that are stored on the surface, realized that CTRANS was using it (to project a shading surface coordinates onto a receiving surface's plane), and I thought I was a bug where the resulting surface was twice as big. 

Could find any test for it, so I wrote one. Turns out I was wrong, the test actually passed. But I left it here, because tests are good.